scheduler/src/cook/mesos/api.clj

```diff
@@ -419,18 +449,43 @@
         :allowed? (fn [ctx]
                     (condp contains? (get-in ctx [:request :request-method])
                       #{:get :delete}
-                      (letfn [(used? [uuid]
-                                (try
-                                  (unused-uuid? (db conn) uuid)
-                                  false
-                                  (catch Exception e
-                                    true)))]
-                        (or (every? used? (::jobs ctx))
-                          [false {::error (str "UUID "
-                                               (str/join
-                                                 \space
-                                                 (remove used? (::jobs ctx)))
-                                               " didn't correspond to a job")}]))
+                      (let [user  (get-in ctx [:request :authorization/user])
+                            uuids (::jobs ctx)
+
+                            datomic (db conn)
+                            used? (fn [uuid]
+                                    (try
+                                      (unused-uuid? datomic uuid)
+                                      false
+                                      (catch Exception e
+                                        true)))
+
+                            ;; Changes each UUID into a map describing
+                            ;; whether it's allowed, and if not, why
+                            ;; it's not allowed.
+                            uuid-filter (fn [uuid]
+                                          (let [job (fetch-job-map (db conn)
+                                                                   fid
+                                                                   uuid)]
+                                            (if (or (nil? job) 
+                                                    (not (is-authorized? user :access job)))
+                                              {:is-allowed false
+                                               :message (str "No job found with UUID " uuid)
```

Comment:
Such a message would leak information to the end user about whether a given UUID exists at all in the system. This might then be used as input to other unrelated processes that are attempting to hack some element of the system but which will only work when given an extant UUID.

This scheme, based on the principle of least privilege, reveals information to the user only on a need-to-know basis. They can't find any job with the UUID they provided, which is all the unauthorized user needs to know about the system state.

This way, a potentially hostile user can't differentiate between the cases where a) the given UUID exists and they are simply not authorized to use it, or b) the UUID just doesn't exist in the system at all. They aren't given a tool for narrowing the "valid job ID" search space by being able to test arbitrary UUIDs for validity.